nemo/collections/asr/models/clustering_diarizer.py

```diff
@@ -260,16 +264,20 @@ def _run_vad(self, manifest_file):
         write_rttm2manifest(AUDIO_VAD_RTTM_MAP, self._vad_out_file)
         self._speaker_manifest_path = self._vad_out_file
 
-    def _run_segmentation(self):
+    def _run_segmentation(self, window=None, shift=None, scale_tag: str = ''):
 
-        self.subsegments_manifest_path = os.path.join(self._speaker_dir, 'subsegments.json')
+        if window is not None:
+            self._speaker_params.window_length_in_sec = window
+        if shift is not None:
+            self._speaker_params.shift_length_in_sec = shift
+
```

Comment:
Read the code, and you will see "window" and "shift" is input of the function, not hydra configration.

Reply:
window and shift are being fed to diarizer from hydra config, which has default non None params. Even if they pass null, by this you are only handling non-None case, what if its None?

Reply:
Or instead of using None for checking, using type for deciding whether it's single or multiscale would be more straightforward. And there will be no need to introduce new variable "window" and "shift" here